plainRFM69.cpp

```diff
@@ -25,268 +107,420 @@ void plainRFM69::setRecommended(){
+     *
+     **/
     return (this->state == RFM69_PLAIN_STATE_RECEIVING);
-    // if we're receiving, we can send.
-    // This is perhaps slightly naive, as a packet might be being received.
-    
-    /*
-    perhaps also check RSSI:
-
-    uint8_t flags1;
-    flags1 = this->getIRQ1Flags();
-    debug_rfm("canSend, flags1: "); debug_rfmln(flags1);
-    return ((this->state == RFM69_PLAIN_STATE_RECEIVING) and ((flags1 & RFM69_IRQ1_RSSI)==0));
-
-    It is more correct, but is it necessary?
-    The IRQ1_RSSI flag is almost always set, dependent on RSSIthresshold...
-    TODO: Investigate this if this causes problems with multiple transmitters.
-    */
-    // perhaps also place a timeout on the sending state??
-    // Just in case an interrupt is missed.
 }
 
-void plainRFM69::sendAddressedVariable(uint8_t address, void* buffer, uint8_t len){
-    this->tx_buffer[1] = address; // set address byte.
-    this->tx_buffer[0] = len+1; // set length, add one for address byte.
-    memcpy(&(this->tx_buffer[2]), buffer, len); // write the payload.
-    this->sendPacket(this->tx_buffer, len+2); // send the payload.
+
+void plainRFM69::sendAddressedVariable(uint8_t address, void* buffer, uint8_t len)
+{
+    this->tx_buffer[1] = address;               // Set address byte.
+    this->tx_buffer[0] = len+1;                 // Set length, add one for address byte.
+    memcpy(&(this->tx_buffer[2]), buffer, len); // Write the payload.
+    this->sendPacket(this->tx_buffer, len+2);   // Send the payload.
 }
-void plainRFM69::sendVariable(void* buffer, uint8_t len){
+
+
+void plainRFM69::sendVariable(void* buffer, uint8_t len)
+{
     this->tx_buffer[0] = len;
     memcpy(&(this->tx_buffer[1]), buffer, len);
     this->sendPacket(this->tx_buffer, len+1);
 }
 
-void plainRFM69::sendAddressed(uint8_t address, void* buffer){
+
+void plainRFM69::sendAddressed(uint8_t address, void* buffer)
+{
     this->tx_buffer[0] = address;
     memcpy(&(this->tx_buffer[1]), buffer, this->packet_length);
     this->sendPacket(this->tx_buffer, this->packet_length+1);
 }
 
-void plainRFM69::send(void* buffer){
+
+void plainRFM69::send(void* buffer)
+{
     this->sendPacket(buffer, this->packet_length);
 }
 
+void plainRFM69::setTxPower(int8_t power_level_dBm, bool enable_boost)
+{
+    /**
+     *
+     *  The code below is based on the knowledge gleaned from this article:
+     *
+     *    https://andrehessling.de/2015/02/07/figuring-out-the-power-level-settings-of-hoperfs-rfm69-hwhcw-modules/
+     *
+     *  Additionally, I studied the following drivers for inspiration and understanding.
+     *
+     *    https://github.com/ahessling/RFM69-STM32
+     *    https://github.com/LowPowerLab/RFM69
+     *    http://www.airspayce.com/mikem/arduino/RadioHead/classRH__RF69.html
+     *
+     **/
+
+    // Determine the hardware type.
+    if (use_HP_module)
+    {
+        // ***** High power module is installed. Things are complicated. *****
+
+        /**
```

Comment:
Parts of this comment block should go into the header file as they provide context, requirements and limitations of the `enable_boost` flag.